### ISSUE TITLE
Standalone VM server crashing with `panic: runtime error: slice bounds out of range [-7::]` in netstorage.go:1236

### ISSUE THREAD

**Issue by Fusl** [VictoriaMetrics/VictoriaMetrics#5733]
### Describe the bug

VictoriaMetrics randomly crashes with `panic: runtime error: slice bounds out of range [-7::]` on this line of code:

https://github.com/VictoriaMetrics/VictoriaMetrics/blob/495fa9800a46ef733fbd1568899ad591bc45e7e5/app/vmselect/netstorage/netstorage.go#L1236

### To Reproduce

Unknown reproduction steps

### Version

victoria-metrics-20240125-090636-heads-master-0-g495fa9800

### Logs

```
panic: runtime error: slice bounds out of range [-7::]

goroutine 13255452 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper.func1()
        github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:242 +0x58
panic({0xada900?, 0x402ea27350?})
        runtime/panic.go:920 +0x26c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage.ProcessSearchQuery(0x4031651ce4?, 0x408124c270, {0x65b9789e, 0x6fc23ac00, {0xb1fcd0, 0x18}})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage/netstorage.go:1236 +0x130c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncNoCache(0xff5d?, 0x4066b88608, {0x4031651ce4, 0xd}, 0xe7?, {0xd8b520, 0x405a37f440?}, 0x4082eb92f0, 0x408124c210, 0x0, ...)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1685 +0x504
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithMetricExpr.func1(0x4000d70020?)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1598 +0x4c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithMetricExpr(0x0, 0x404dba38c0, {0x4031651ce4, 0xd}, 0xcd6da0, {0xd8b520, 0x405a37f440}, 0x4082eb92f0, 0x408124c210, 0x10000000004156c8?)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1634 +0x348
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithoutAt(0xe9ee2d5bbcd8?, 0x404dba38c0, {0x4031651ce4?, 0x40c2a4?}, 0x40bc324848?, {0xd8b520, 0x405a37f440}, 0x430a9fc640, 0x408124c210)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:862 +0x42c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFunc(0x40002d0b00?, 0x4000125080?, {0x4031651ce4?, 0x3?}, 0x74656d013101746e?, {0xd8b520?, 0x405a37f440?}, 0x65b97880?, 0x7375746174730173?)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:799 +0x378
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalAggrFunc(0x4031651ce0?, 0x404dba38c0, 0x405a37f440)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:399 +0x45c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExprInternal(0x40bc324c18?, 0x404dba38c0, {0xd8b520?, 0x405a37f440?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:325 +0x31c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExpr(0x119d560?, 0x61?, {0xd8b520?, 0x405a37f440?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:267 +0x1f8
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.Exec(0x40bc324f18?, 0x404dba38c0, {0x4117344770, 0x61}, 0x0)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/exec.go:67 +0x164
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/prometheus.queryRangeHandler(0x0, {0xb11eea?, 0x4?, 0x11a09e0?}, {0xd8e800, 0x4000f00e70}, {0x4117344770, 0x61}, 0x18d5c633cc0, 0x18d5c7eb400, ...)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/prometheus/prometheus.go:908 +0x470
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/prometheus.QueryRangeHandler(0xd8e800?, {0x4000f00e70?, 0x408f443900?, 0x11a09e0?}, {0xd8e800, 0x4000f00e70}, 0xffffffffffffffff?)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/prometheus/prometheus.go:860 +0x1e0
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect.RequestHandler({0xd8e800, 0x4000f00e70}, 0x408f443900)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/main.go:220 +0x1ddc
main.requestHandler({0xd8e800?, 0x4000f00e70}, 0x408f443900)
        github.com/VictoriaMetrics/VictoriaMetrics/app/victoria-metrics/main.go:133 +0x260
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper(0x40002d0a20, {0xd8ee00?, 0x4035748000}, 0x408f443900, 0xcd6888)
        github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:367 +0x1004
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.gzipHandler.func1({0xd8ee00?, 0x4035748000?}, 0x4063809901?)
        github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:201 +0x3c
net/http.HandlerFunc.ServeHTTP(0x1134c00?, {0xd8ee00?, 0x4035748000?}, 0x4?)
        net/http/server.go:2136 +0x38
github.com/klauspost/compress/gzhttp.NewWrapper.func1.1({0xd8e5f0, 0x4094fe1880}, 0x40002d09b0?)
        github.com/klauspost/compress@v1.17.4/gzhttp/compress.go:495 +0x2ac
net/http.HandlerFunc.ServeHTTP(0x10?, {0xd8e5f0?, 0x4094fe1880?}, 0x4094fe1880?)
        net/http/server.go:2136 +0x38
net/http.serverHandler.ServeHTTP({0xd8d020?}, {0xd8e5f0?, 0x4094fe1880?}, 0x6?)
        net/http/server.go:2938 +0xbc
net/http.(*conn).serve(0x4037cf79e0, {0xd8f490, 0x4112cadef0})
        net/http/server.go:2009 +0x518
created by net/http.(*Server).Serve in goroutine 1931
        net/http/server.go:3086 +0x4cc
```

### Used command-line flags

```
-http.pathPrefix="/vm_<redacted>"
-httpListenAddr=":8428"
-influxMeasurementFieldSeparator="."
-insert.maxQueueDuration="15m0s"
-maxConcurrentInserts="256"
-memory.allowedPercent="75"
-retentionPeriod="10y"
-search.disableCache="false"
-search.latencyOffset="0s"
-search.maxConcurrentRequests="64"
-search.maxQueryDuration="30s"
-search.maxQueueDuration="30s"
-search.maxSamplesPerSeries="100000000"
-search.maxUniqueTimeseries="10000000"
-storageDataPath="/storage"
```

**Comment by valyala** [comment]
@Fusl , could you verify whether the issue exists at [v1.97.0 release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.97.0)?

**Comment by Fusl** [comment]
Just deployed `victoria-metrics-20240130-210219-tags-v1.97.0-0-gfcc8b14f8` onto our monitoring stack. I don't have a good reproduction case but noticed it happens more often than just every few hours when we have an increased new series and churn rate. So far 1.97.0 has been running just fine for the past few minutes, I'll see if I can cause this issue to also happen on this version.

**Comment by Fusl** [comment]
Crashes again twice in a row:

```
panic: runtime error: slice bounds out of range [-6::]

goroutine 5584260 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage.ProcessSearchQuery(0x4107f8ae91?, 0x40b8e83320, {0x65b9c273, 0x6fc23ac00, {0xaf58de, 0x18}})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage/netstorage.go:1236 +0x130c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncNoCache(0x40b29d5718?, 0x4124a6d878, {0x4107f8ae91, 0x4}, 0x40b29d5738?, {0xd5d660, 0x40ae6b5980?}, 0x40ae640ed0, 0x40b8e832c0, 0x0, ...)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1698 +0x504
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithMetricExpr.func1(0x400057c668?)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1611 +0x4c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithMetricExpr(0x0, 0x400114c780, {0x4107f8ae91, 0x4}, 0xcac3d0, {0xd5d660, 0x40ae6b5980}, 0x40ae640ed0, 0x40b8e832c0, 0x40?)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1647 +0x348
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithoutAt(0xf53c7308ac64337?, 0x400114c780, {0x4107f8ae91?, 0x40c2a4?}, 0x400019d010?, {0xd5d660, 0x40ae6b5980}, 0x41e28a7340, 0x40b8e832c0)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:862 +0x42c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFunc(0x40de600cf0?, 0x4000402e40?, {0x4107f8ae91?, 0x3?}, 0x402bcbf3a8?, {0xd5d660?, 0x40ae6b5980?}, 0x78baf4?, 0x402bcbf3e8?)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:799 +0x378
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalAggrFunc(0x4107f8ae8d?, 0x400114c780, 0x40ae6b5980)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:399 +0x45c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExprInternal(0x402bcbf688?, 0x400114c780, {0xd5d660?, 0x40ae6b5980?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:325 +0x31c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExpr(0x4056995000?, 0x40ae51be00?, {0xd5d660?, 0x40ae6b5980?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:267 +0x1f8
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.execBinaryOpArgs.func2()
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:502 +0x74
created by github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.execBinaryOpArgs in goroutine 5583953
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:500 +0x51c
```
```
panic: runtime error: slice bounds out of range [-4::]

goroutine 97846 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper.func1()
        github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:242 +0x58
panic({0xab2f20?, 0x42405b9c68?})
        runtime/panic.go:920 +0x26c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage.ProcessSearchQuery(0x40bba3c2d6?, 0x411a8e8030, {0x65b9c2d2, 0x6fc23ac00, {0xaf58de, 0x18}})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage/netstorage.go:1236 +0x130c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncNoCache(0x0?, 0x42924197c8, {0x40bba3c2d6, 0xd}, 0x427d0a3688?, {0xd5d660, 0x4064f1efc0?}, 0x40b2c31380, 0x40e9ceff80, 0x0, ...)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1698 +0x504
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithMetricExpr.func1(0x4000b94060?)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1611 +0x4c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithMetricExpr(0x0, 0x4133a695c0, {0x40bba3c2d6, 0xd}, 0xcac4d0, {0xd5d660, 0x4064f1efc0}, 0x40b2c31380, 0x40e9ceff80, 0x10000000004156c8?)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1647 +0x348
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithoutAt(0xe9e90609a288?, 0x4133a695c0, {0x40bba3c2d6?, 0x40c2a4?}, 0x427d0a3a08?, {0xd5d660, 0x4064f1efc0}, 0x42a8d89f00, 0x40e9ceff80)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:862 +0x42c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFunc(0x41a8e9a9a0?, 0x40004a4720?, {0x40bba3c2d6?, 0x3?}, 0x427d0a3af8?, {0xd5d660?, 0x4064f1efc0?}, 0x8ff914?, 0x427d0a3bb8?)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:799 +0x378
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalAggrFunc(0x40bba3c2d2?, 0x4133a695c0, 0x4064f1efc0)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:399 +0x45c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExprInternal(0x4?, 0x4133a695c0, {0xd5d660?, 0x4064f1efc0?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:325 +0x31c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExpr(0x40bcbaca00?, 0x414500?, {0xd5d660?, 0x4064f1efc0?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:267 +0x1f8
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.execBinaryOpArgs(0x40bba3c33a?, 0x4133a695c0, {0xd5d660?, 0x4064f1efc0}, {0xd5d660, 0x4064f1f140}, 0x40bcbaca00)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:539 +0xb8
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalBinaryOp(0x40bba3c33a?, 0xaeb915?, 0x40bcbaca00)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:441 +0xfc
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExprInternal(0x4?, 0x4133a695c0, {0xd5d6e0?, 0x40bcbaca00?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:331 +0x258
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExpr(0x40bcbacaa0?, 0x73746174735f7373?, {0xd5d6e0?, 0x40bcbaca00?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:267 +0x1f8
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.execBinaryOpArgs(0x40bba3c3a3?, 0x4133a695c0, {0xd5d6e0?, 0x40bcbaca00}, {0xd5d6c0, 0x4000ad9650}, 0x40bcbacaa0)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:539 +0xb8
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalBinaryOp(0x40bba3c3a3?, 0xaeb915?, 0x40bcbacaa0)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:441 +0xfc
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExprInternal(0x7463656c65732f64?, 0x4133a695c0, {0xd5d6e0?, 0x40bcbacaa0?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:331 +0x258
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExpr(0x40bcbacb40?, 0x75622d7265747361?, {0xd5d6e0?, 0x40bcbacaa0?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:267 +0x1f8
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.execBinaryOpArgs(0x40bba3c3ae?, 0x4133a695c0, {0xd5d6e0?, 0x40bcbacaa0}, {0xd5d6c0, 0x4000ad96c8}, 0x40bcbacb40)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:539 +0xb8
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalBinaryOp(0x40bba3c3ae?, 0xaeb915?, 0x40bcbacb40)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:441 +0xfc
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExprInternal(0x4278f3ac18?, 0x4133a695c0, {0xd5d6e0?, 0x40bcbacb40?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:331 +0x258
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExpr(0x1157c00?, 0xe1?, {0xd5d6e0?, 0x40bcbacb40?})
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:267 +0x1f8
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.Exec(0x4278f3af18?, 0x4133a695c0, {0x4110b184b0, 0xe1}, 0x0)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/exec.go:67 +0x164
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/prometheus.queryRangeHandler(0x0, {0xae7c8b?, 0x4?, 0x115ab60?}, {0xd606f8, 0x412b381680}, {0x4110b184b0, 0xe1}, 0x18d5d32b220, 0x18d5da08f20, ...)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/prometheus/prometheus.go:914 +0x470
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/prometheus.QueryRangeHandler(0xd606f8?, {0x412b381680?, 0x4231837300?, 0x115ab60?}, {0xd606f8, 0x412b381680}, 0xffffffffffffffff?)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/prometheus/prometheus.go:866 +0x1e0
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect.RequestHandler({0xd606f8, 0x412b381680}, 0x4231837300)
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/main.go:220 +0x1ddc
main.requestHandler({0xd606f8?, 0x412b381680}, 0x4231837300)
        github.com/VictoriaMetrics/VictoriaMetrics/app/victoria-metrics/main.go:133 +0x260
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper(0x40011784f0, {0xd60cf8?, 0x406ce8e500}, 0x4231837300, 0xcabe78)
        github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:367 +0x1004
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.gzipHandler.func1({0xd60cf8?, 0x406ce8e500?}, 0x427bb1a901?)
        github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:201 +0x3c
net/http.HandlerFunc.ServeHTTP(0x10ef560?, {0xd60cf8?, 0x406ce8e500?}, 0x4?)
        net/http/server.go:2136 +0x38
github.com/klauspost/compress/gzhttp.NewWrapper.func1.1({0xd604e8, 0x423fa09260}, 0x41a8e9a850?)
        github.com/klauspost/compress@v1.17.5/gzhttp/compress.go:495 +0x2ac
net/http.HandlerFunc.ServeHTTP(0x10?, {0xd604e8?, 0x423fa09260?}, 0x423fa09260?)
        net/http/server.go:2136 +0x38
net/http.serverHandler.ServeHTTP({0xd5ef40?}, {0xd604e8?, 0x423fa09260?}, 0x6?)
        net/http/server.go:2938 +0xbc
net/http.(*conn).serve(0x415afa34d0, {0xd61390, 0x40c7f97c50})
        net/http/server.go:2009 +0x518
created by net/http.(*Server).Serve in goroutine 5701
        net/http/server.go:3086 +0x4cc
2024-01-31T03:47:25.542Z        info    VictoriaMetrics/lib/logger/flag.go:12   build version: victoria-metrics-20240130-210219-tags-v1.97.0-0-gfcc8b14f8
```

**Comment by Fusl** [comment]
Unsure how helpful or wrong this information is but from a quick dive into the commit history of this file it's quite possible that this bug was introduced with [this `app/vmselect/netstorage: use []blockRef from blockRefPool in order to reduce memory allocations` commit](https://github.com/VictoriaMetrics/VictoriaMetrics/commit/5dd37ad836ef1381ce7d31e9efdd68d7aae984b5)

**Comment by valyala** [comment]
@Fusl , thanks for the provided `panic` stack traces for `v1.97.0` and for figuring out the commit, which could cause the panic! It should be fixed in the commit db4623efc26b6892fcfad739ab8fec371c9baa90 . Could you build VictoriaMetrics from this commit according to [these docs](https://docs.victoriametrics.com/#how-to-build-from-sources) and verify whether the panic is gone there? This commit will be included in the next bugfix release.

**Comment by Fusl** [comment]
Running `victoria-metrics-20240131-195254-heads-master-0-gc8a96ac24` now, going to report again whether I see more crashes happening or if that fixes it.

**Comment by valyala** [comment]
[The bugfix](https://github.com/VictoriaMetrics/VictoriaMetrics/commit/db4623efc26b6892fcfad739ab8fec371c9baa90) has been included in [v1.97.1 release](https://docs.victoriametrics.com/CHANGELOG.html). Closing the issue as fixed then.

@Fusl , please re-open the issue if it still triggers in `v1.97.1` on your workload.
